File: MaybeNewSolutions/题目1040：Prime Number.java

import java.util.Scanner;

/**
 * @ClassName: Main1046
 * @Description: TODO
 * @author wangzq
 * @date 2013-8-7 下午04:00:39
 * 
 * @version 3.0.0
 */
public class Main {
	/*
	 * 1040
	 */
	public static void main(String[] args) {
		int array[] = getPrimeArray();
		Scanner scanner = new Scanner(System.in);
		while (scanner.hasNext()) {
			int K = scanner.nextInt();
			int count = 0;
			for (int i = 0; i < array.length; i++) {
				if (array[i]!=0) {
					count++;
				}
				if (count == K) {
					System.out.println(i);
					break;
				}
			}
		}
	}

	private static int[] getPrimeArray() {
		int length = 200000;
		int array[] = new int[length];
		for (int i = 0; i < length; i++) {
			if (i%2!=0) {
				array[i] = 1;
			}else {
				array[i] = 0;
			}
		}
		array[0] = 0;
		array[1] = 0;
		array[2] = 1;
		for (int i = 2; i*i < length; i++) {
			if (array[i]!=0) {
				for (int j = 2*i; j < length; j+=i) {
					array[j] = 0;
				}
			}
		}
		return array;
	}
}

/**************************************************************
	Problem: 1040
	User: wzqwsrf
	Language: Java
	Result: Accepted
	Time:120 ms
	Memory:18200 kb
****************************************************************/


                        

File: C++/题目1040：Prime Number.cpp

// 题目1040：Prime Number

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:52
 * @url:http://ac.jobdu.com/problem.php?pid=1040
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/35781195
 */

#include <stdio.h>

const int len = 200000;
int array[len];
int i, j, k;

void initArray() {
    for (i = 1; i < len; i++) {
        if (i % 2 == 0) {
            array[i] = 0;
        } else {
            array[i] = 1;
        }
    }
    array[1] = 0;
    array[2] = 1;
    for (i = 3; i < len; i++) {
        if (array[i] == 1) {
            for (j = 2 * i; j < len; j += i) {
                array[j] = 0;
            }
        }
    }
}

int main() {
    initArray();
    while (scanf("%d", &k) != EOF) {
        int count = 0;
        for (int i = 1; i < len; i++) {
            if (array[i] == 1) {
                count++;
            }
            if (count == k) {
                printf("%d\n", i);
                break;
            }
        }
    }
    return 0;
}
/**************************************************************
Problem: 1040
User: wangzhenqing
Language: C++
Result: Accepted
Time:10 ms
Memory:1800 kb
****************************************************************/


                        

